raiden/transfer/token_network.py

```diff
@@ -30,50 +30,78 @@
-            channel_state = ids_to_channels[node_address].get(state_change.channel_identifier)
+        partner_to_channelids = token_network_state.partneraddresses_to_channelidentifiers[
+            channel_state.partner_state.address
+        ]
+
+        channel_identifier = state_change.channel_identifier
+        if result.new_state is None:
+            del ids_to_channels[channel_state.our_state.address][channel_identifier]
+            partner_to_channelids.remove(channel_identifier)
         else:
-            lc_address = views.get_lc_address_by_channel_id_and_partner(token_network_state, node_address,
-                                                                    state_change.canonical_identifier)
-            node_address = lc_address
-            if lc_address in token_network_state.channelidentifiers_to_channels:
-                channel_state = token_network_state.channelidentifiers_to_channels[lc_address].get(
-                    state_change.canonical_identifier.channel_identifier)
-        if channel_state:
-            result = channel.state_transition(
-                channel_state=channel_state,
-                state_change=state_change,
-                block_number=block_number,
-                block_hash=block_hash,
-            )
-
-            partner_to_channelids = token_network_state.partneraddresses_to_channelidentifiers[
-                channel_state.partner_state.address
-            ]
+            ids_to_channels[channel_state.our_state.address][channel_identifier] = result.new_state
 
-            channel_identifier = state_change.channel_identifier
-            if result.new_state is None:
-                del ids_to_channels[node_address][channel_identifier]
-                partner_to_channelids.remove(channel_identifier)
-            else:
-                ids_to_channels[node_address][channel_identifier] = result.new_state
+        events.extend(result.events)
+    return TransitionResult(token_network_state, events)
 
-            events.extend(result.events)
 
-    return TransitionResult(token_network_state, events)
+#  TODO: We should find another approach for ths function. It's "a bit" confusing
+def get_channels_to_dispatch_statechange(
+    participant_address: AddressHex,
+    state_change: StateChangeWithChannelID,
+    token_network_state: TokenNetworkState
+) -> List[NettingChannelState]:
+    channel_states = []
+    ids_to_channels = token_network_state.channelidentifiers_to_channels
+
+    if participant_address in ids_to_channels:
+        channel_state = ids_to_channels[participant_address].get(state_change.channel_identifier)
+        channel_states.append(channel_state)
+        if channel_state.both_participants_are_light_clients:
+            partner_channel = ids_to_channels[channel_state.partner_state.address]
+            partner_channel_state = partner_channel.get(channel_state.identifier)
+            channel_states.append(partner_channel_state)
+    else:
```

Comment:
the code structure looks good, but i don't understand what this is is evaluating.

what does it mean that `participant_address in ids_to_channels == True`?

can we maybe express this boolean condition in a clearer, more business logic-oriented way?

at first i thought this was about light clients, but it seems both branches talk about LCs, so i'm not so sure now

Reply:
This was part of the discussion we had yesterday.

It's seems like both branches can be lc or full nodes, even the else branch can be a full node. I found the naming wrong if that's the case

I understand that:
If the participant is our (we are the full node owner of the address, or the corresponding lc is handled by us) then it goes to the first branch.

Else, it check if we have a channel with that address as a partner.

I think I know a few ways to improve this, but we don't even have an agreement on what does the 'else' does.

Reply:
Agree.

Reply:
the other option I thought was renaming `ids_to_channels` to something like `our_channels`, so `participant_address in our_channels` would be a more explicit way of saying 'Is this participant address a side of some channel we own?'. 

this  would apply both to this conditio and to the one in line 99